frontend-react/src/components/USLink.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { PropsWithChildren } from "react";
+import { Link, NavLink } from "react-router-dom";
+
+// Known issue with the `PropsWithChildren generic in React 18,
+// so I wrote this in a way that we can just remove `<any>` and be
+// okay if we update our React version.
+interface USNavLinkProps extends PropsWithChildren<any> {
+    href: string;
+    className?: string;
+    activeClassName?: string;
+}
+type USLinkProps = Omit<USNavLinkProps, "activeClassName">;
+
+/** A single link for rendering standard links */
+export const USLink = ({ children, href, className }: USLinkProps) => (
```

Comment:
Just a thought: for generic components like this, would it be useful if we have a single component with an enum-able `variant` (or whatever we could call it) prop as opposed to multiple components?  I feel as though it might get a little hard to maintain if we have a separate component for every state 🤔

Reply:
I think I originally did do something like that, but I think React bloggers are getting to me. I was reading the React docs "Inheritance vs Composition", and there was a section about "Specializations" that used this pattern and I liked it :)

https://reactjs.org/docs/composition-vs-inheritance.html#specialization

Reply:
I'll put you on the actual refactor PR and we can see how things ended up and whether we like it or not. Should have the PR ready later tonight! I also changed some of the component code to use more robust types for stuff like "noreferrer noopener"

Reply:
Sounds good to me!  I'm still a bit wary about having a separate class for every visual state since right now we're basically just changing the `className` prop (the `to` versus `href` could be abstracted away by checking against react-router context).  My main concern is the maintenance of these separate components for every state -- making a global change (e.g., adding a new prop) for the link components would require us to update each one.  

(There's also the potential for byte bloat in the resultant bundle since each of these would get transpiled into its own `createElement` with its own JSX, but that's significantly less important than longterm maintenance in my mind!)

Reply:
Ah, yes! The updated branch (checkout experience/7730/whatever-i-named-the-new-one) and take a peek at that USLink.tsx file :) Solves the concern by relying again on React's definition of AnchorHTMLAttributes and everything should be more centralized in USLink as it's the only one adapting props to a Link component from react-router-dom

Reply:
Hmmmm my feeling here is that this particular use case still doesn't need more than one component with different variants because all of this is slightly different output with the same underlying responsibilities:
1. Determine if rendering within a router context
2. Determine if rendering an absolute url
3. Apply a different `className` prop based on the variant
4. Apply different attributes based on the variant (if applicable)
5. Forward native HTML attributes (`target`, `rel`, et cetera) to the element (like even though it doesn't make a lot of sense in most cases, Link/NavLink can also open new tabs so who are we to restrict the consuming component's needs :P)

Putting it in Storybook terms, these would just be different stories of the same component rather than stories of different components 🤓

But aside from this, I thiiiiiiiink there's an issue with the code in the new branch since Link/NavLink would end up like `<a href="https://reportstream.cdc.gov/https://www.google.com">whatever</a>` when it's rendered if passed an absolute url.  I'm thinking we should handle that properly under the hood so we can use the same component(s) all over the place.

Reply:
>Determine if rendering within a router context
>Determine if rendering an absolute url

This decision is much easier made by a human, and the less complex logic packed into a component, the lest testing required. Declaring `USExtLink` and `USNavLink` means we can declare their necessary styles, props, and base component rather than infer them from a `variant`.

>Apply a different className prop based on the variant
>Apply different attributes based on the variant (if applicable)

This, again, could be handled by `variant` parsing logic but why? Putting it into a new declaration that proxies the base `USLink` and pre-applies the right styles and props is less dynamic logic to test and still maintains the practice of reusing code at the `USLink` level. It's also a lot easier read than parsing through a `variant` switch statement, then seeing what else that may affect.

>Forward native HTML attributes (target, rel, et cetera) to the element

This is `USLink`'s purpose; each of the specializations (sans `USNavLink` as it relies on a different base for extended Router functionality like `activeClassName`) declaratively applies the necessary special attributes, and pipes the rest down to the base component to also be applied. 

---

`Link` was an issue, so I've changed it back to an `a` tag, but due to its use case, I **don't** think we should be considering any external `NavLink`s. The only true added utility that we have with it right now is using its `activeClassName` prop and inner logic to give `ReportStreamHeader` navigation items the proper active styles (which, as a side effect, this refactor will actually _fix_ considering we don't use the active styles right now like we should).